components/feature/tabs/src/main/java/mozilla/components/feature/tabs/ext/BrowserState.kt

```diff
@@ -36,3 +37,15 @@ internal fun BrowserState.toTabList(
 
     return Pair(tabStates, selectedTabId)
 }
+
+/**
+ * Converts the tabs in [BrowserState], using [tabPartitionsFilter], to a [TabPartition] object.
+ *
+ * This implementation is use to help observe changes to the [BrowserState] tabs when only a select
+ * few properties have changed.
+ */
+internal fun BrowserState.toTabParition(
+    tabPartitionsFilter: (Map<String, TabPartition>) -> TabPartition? = { null }
+): TabPartition? {
+    return tabPartitionsFilter(tabPartitions)
```

Comment:
This extension doesn't seem to do anything. 🤔 Maybe just use it directly from the `defaultTabPartitionsFilter` in `TabsFeature`?

Reply:
Maybe wrap this into the `toTabs` above so we don't need to construct a pair later on?

Reply:
This is used in the Fenix side to filter to the search term group partition.  I'll update the Fenix PR soon so you can take a look from both sides.  Thanks

Reply:
I've moved tabPartitionsFilter out of the extension.  I tried to merge it to `toTabs` but it made the code a bit more confusing.  I'll leave it separate.  Thanks,